pydantic/_analyzed_type.py

```diff
@@ -0,0 +1,217 @@
+        self.core_schema = core_schema
+        self.core_config = core_config
+        self._validator: SchemaValidator | None = None
+        self._schema: SchemaSerializer | None = None
+
+    @property
+    def validator(self) -> SchemaValidator:
+        if self._validator is None:
+            self._validator = SchemaValidator(self.core_schema, self.core_config)
+        return self._validator
+
+    @property
+    def serializer(self) -> SchemaSerializer:
+        if self._schema is None:
+            self._schema = SchemaSerializer(self.core_schema, self.core_config)
+        return self._schema
+
+    def validate_python(self, __object: Any) -> T:
+        return self.validator.validate_python(__object)
+
+    def validate_json(self, __data: str | bytes) -> T:
+        return self.validator.validate_json(__data)
+
+    def dump_python(
+        self,
+        __instance: T,
+        *,
+        mode: Literal['json', 'python'] = 'python',
+        include: IncEx | None = None,
+        exclude: IncEx | None = None,
+        by_alias: bool = False,
+        exclude_unset: bool = False,
+        exclude_defaults: bool = False,
+        exclude_none: bool = False,
+        round_trip: bool = False,
+        warnings: bool = True,
+    ) -> Any:
+        return self.serializer.to_python(
+            __instance,
+            mode=mode,
+            by_alias=by_alias,
+            include=include,
+            exclude=exclude,
+            exclude_unset=exclude_unset,
+            exclude_defaults=exclude_defaults,
+            exclude_none=exclude_none,
+            round_trip=round_trip,
+            warnings=warnings,
+        )
+
+    def dump_json(
+        self,
+        __instance: T,
+        *,
+        indent: int | None = None,
+        include: IncEx | None = None,
+        exclude: IncEx | None = None,
+        by_alias: bool = False,
+        exclude_unset: bool = False,
+        exclude_defaults: bool = False,
+        exclude_none: bool = False,
+        round_trip: bool = False,
+        warnings: bool = True,
+    ) -> bytes:
+        return self.serializer.to_json(
+            __instance,
+            indent=indent,
+            include=include,
+            exclude=exclude,
+            by_alias=by_alias,
+            exclude_unset=exclude_unset,
+            exclude_defaults=exclude_defaults,
+            exclude_none=exclude_none,
+            round_trip=round_trip,
+            warnings=warnings,
+        )
+
+    @overload
+    @staticmethod
+    def analyze_type(__type: type[T], *, config: CoreConfig | None = None, parent_depth: int = 2) -> AnalyzedType[T]:
+        ...
+
+    @overload
+    @staticmethod
+    def analyze_type(__type: Any, *, config: CoreConfig | None = None, parent_depth: int = 2) -> AnalyzedType[Any]:
+        ...
+
+    @staticmethod
+    def analyze_type(__type: Any, *, config: CoreConfig | None = None, parent_depth: int = 2) -> AnalyzedType[Any]:
```

Comment:
I wonder if we should change this argument name to `_parent_depth` to discourage users from trying to use it lol

Reply:
I'm open to accepting a namespace or a call depth to get it from instead of just the call depth. Then `BaseModel` could use the fact that it knows it's own module and whatnot.